packages/integrations/preact/server.js

```diff
@@ -4,26 +4,74 @@ import StaticHtml from './static-html.js';
 
 const slotName = (str) => str.trim().replace(/[-_]([a-z])/g, (_, w) => w.toUpperCase());
 
+let originalConsoleError;
+let runningChecks = 0;
+
 function check(Component, props, children) {
 	if (typeof Component !== 'function') return false;
 
 	if (Component.prototype != null && typeof Component.prototype.render === 'function') {
 		return BaseComponent.isPrototypeOf(Component);
 	}
 
+	installConsoleErrorFilter();
+
+	runningChecks++;
```

Comment:
Do you think wrapping these up as `useConsoleFilter()` and the decrement as `finishUsingConsoleFilter()` could help keep that logic separate? Other than that, this LGTM!

Reply:
Great idea! :)